### ISSUE TITLE
Tasks: Configure Task lists only first folder in workspace

### ISSUE THREAD

**Issue by szekelyisz** [microsoft/vscode-cmake-tools#3232]
### Brief Issue Summary

I have several folders linked into my vscode workspace, mostly TypeScript projects and also a CMake one. When I run `Tasks: Configure Task`, the only choice to create a task with the CMake template is for the top and first folder. I had to drag my CMake project to the top to be given the right choice when creating a task, targeting my CMake project instead of a TypeScript one.

I would expect all templates be available to all folders in the workspace just like the way the NodeJS task generator works.

**Comment by benmcmorran** [comment]
Thanks for the suggestion! Your suggested change makes sense to me. We may not be able to prioritize it immediately but would welcome a pull request if someone else gets to it first.

**Comment by hippo91** [comment]
@szekelyisz i would like to tackle this issue but cannot reproduce clearly the problem. Can you give me a reproducer or at least at screencast illustrating it?

**Comment by Yingzi1234** [comment]
@gcampbell-msft We ran into a problem when trying to reproduce the user's question. After running the command A, we did not see the selected folder name within the list within the command line. Is there any way we can see exactly which folder was run?

![Image](https://github.com/user-attachments/assets/b0bc6567-ed0d-4280-b2b7-55c62fe24d25)

**Customer:**
![Image](https://github.com/user-attachments/assets/bdc15e12-cd23-453d-9a53-3b400dbeca29)

**Comment by szekelyisz** [comment]
> @gcampbell-msft We ran into a problem when trying to reproduce the user's question. After running the command A, we did not see the selected folder name within the list within the command line. Is there any way we can see exactly which folder was run?

Did you try with multiple folders in your workspace? I think the folder name is only shown then.

**Comment by Yingzi1234** [comment]
@szekelyisz Yes I did create two different folders in the workspace, I'm not sure if the cause of your error is related to the configure project that was selected at the very beginning after run command CMake: configure, my reproduction steps are below, you can take a look.
1. Open a test file containing multiple folders
2. Click F1 to run command `CMake: Configure`
3. Choose one of the test folders
4. Click F1 to run command `Task: configure task`
![Image](https://github.com/user-attachments/assets/b974db44-a8e5-4497-9383-041429c5cb6b)

**Comment by szekelyisz** [comment]
> @szekelyisz Yes I did create two different folders in the workspace, I'm not sure if the cause of your error is related to the configure project that was selected at the very beginning after run command CMake: configure, my reproduction steps are below, you can take a look.
> 1. Open a test file containing multiple folders
> 2. Click F1 to run command `CMake: Configure`
> 3. Choose one of the test folders
> 4. Click F1 to run command `Task: configure task`

No that's not right. You have one folder in your workspace with two subfolders.

Start a new workspace and open your first folder. Then right click in the tree view and select _Add folder to workspace_. Now try `Task: configure task`.

**Comment by Yingzi1234** [comment]
@szekelyisz Thank you for your reply! We can reproduce this issue based the above information. We have been changed this issue to bug, please be patient with the results of our investigation and we will notify you of any updates as soon as they are available!

@gcampbell-msft We can reproduce the issue on latest CMake Tools version 1.20.32(pre-release) and change the Label to ‘**Bug’** for now, if it is wrong could you please change it to the correct label, the reproduction steps are detailed below. Thanks!

**Issue:** "Tasks: Configure Task" lists only first folder in workspace

**More info:**
This issue is only reproduced on Windows platforms

**Repro steps:**
1. Open the project containing a folder
2. Right click the Explorer page and click "Add folder to workspace" button
3. Select 'Add' another folder
4. Click F1 to run command "Tasks: configure task"
5. Check the folder name in the list
6. Move the other folder as the first folder and repeat steps 4-5

**Actual results:**
The task list shows only the name of the first folder
![Image](https://github.com/user-attachments/assets/36b8eabb-b0af-4773-b8cc-0a0ec157d89e)

**Comment by hippo91** [comment]
@szekelyisz correct me if i am wrong but the problem occurs also with the `Tasks: Run Task` command. Isn't it?

**Comment by hippo91** [comment]
@szekelyisz if you could test the branch of #4223 and tell me if everything work as expected, it would be nice!

**Comment by Yingzi1234** [comment]
@hippo91 Theoretically we think it's a problem because `Tasks: run task`'s task depends on the option selected by running `Tasks: configure task` in the previous step
> [@szekelyisz](https://github.com/szekelyisz) correct me if i am wrong but the problem occurs also with the `Tasks: Run Task` command. Isn't it?
